tm2/pkg/crypto/keys/client/verify.go

```diff
@@ -39,63 +146,227 @@ func (c *VerifyCfg) RegisterFlags(fs *flag.FlagSet) {
 	var msg []byte
 
 	// read document to sign
-	if docpath == "" { // from stdin.
+	if cfg.DocPath == "" { // from stdin.
 		msgstr, err := io.GetString(
 			"Enter document to sign.",
 		)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		msg = []byte(msgstr)
 	} else { // from file
-		msg, err = os.ReadFile(docpath)
+		var err error
+		msg, err = os.ReadFile(cfg.DocPath)
 		if err != nil {
-			return err
+			return nil, err
 		}
 	}
 
-	// validate document to sign.
-	// XXX
+	// Unmarshal transaction Amino JSON
+	var tx std.Tx
+	if err := amino.UnmarshalJSON(msg, &tx); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal transaction, %w", err)
+	}
+	return &tx, nil
+}
 
-	// verify signature.
-	err = kb.Verify(name, msg, sig)
-	if err == nil {
-		io.Println("Valid signature!")
+func getSignature(cfg *VerifyCfg, tx *std.Tx) ([]byte, error) {
+	// Exclude -sigpath and -signature flags set at the same time.
+	if cfg.SigPath != "" && cfg.Signature != "" {
+		return nil, errors.New("only one of -sigpath or -signature flags can be set")
+	}
+
+	// from -sigpath flag
+	if cfg.SigPath != "" {
+		sigbz, err := os.ReadFile(cfg.SigPath)
+		if err != nil {
+			return nil, err
+		}
+
+		// Unmarshal transaction Amino JSON
+		var sig std.Signature
+		if err := amino.UnmarshalJSON(sigbz, &sig); err != nil {
+			return nil, fmt.Errorf("unable to unmarshal signature, %w", err)
+		}
+
+		if sig.Signature == nil {
+			return nil, errors.New("no signature found in the signature file")
+		}
+
+		return sig.Signature, nil
+	}
+
+	// from -signature flag
+	if cfg.Signature != "" {
+		sig, err := base64.StdEncoding.DecodeString(cfg.Signature)
+		if err != nil {
+			return nil, fmt.Errorf("unable to decode signature, %w", err)
+		}
+		return sig, nil
+	}
+
+	// default: from tx
+	if len(tx.Signatures) > 0 {
+		return tx.Signatures[0].Signature, nil
+	}
+
+	return nil, errors.New("no signature found in the transaction")
+}
+
+func getTxParameters(ctx context.Context, cfg *VerifyCfg, info keys.Info, io commands.IO) {
+	// Query the chain if -offline=false and account number or sequence are equal to 0.
+	if !cfg.Offline && (cfg.AccountNumber == 0 || cfg.AccountSequence == 0) {
+		if !cfg.RootCfg.BaseOptions.Quiet {
+			io.Println("Querying account from chain...")
+		}
+		// Query the account from the chain.
+		baseAccount, err := queryBaseAccount(ctx, cfg, info)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Warning: could not query account from chain, use default values: %v\n", err)
+		} else {
+			if cfg.AccountNumber == 0 {
+				if !cfg.RootCfg.BaseOptions.Quiet {
+					io.Printf("account-number set to %d\n", baseAccount.AccountNumber)
+				}
+				cfg.AccountNumber = baseAccount.AccountNumber
+			}
+			if cfg.AccountSequence == 0 {
+				if !cfg.RootCfg.BaseOptions.Quiet {
+					io.Printf("account-sequence set to %d\n", baseAccount.Sequence)
+				}
+				cfg.AccountSequence = baseAccount.Sequence
+			}
```

Comment:
Are there cases where you might want to fetch only one value (account sequence / number) and manually set the other one? 
It feels odd in terms of flow, either the user relies on `gnokey` to fetch the values on-chain or set them both manually, right?
WDYT @zivkovicmilos?